agilent_infiniium_scope.py

```diff
@@ -31,39 +31,39 @@
 """
 
 
-
 from labrad import types as T, util
 from labrad.server import setting
 from labrad.gpib import GPIBManagedServer, GPIBDeviceWrapper
 from twisted.internet.defer import inlineCallbacks, returnValue
 import labrad.units as U
 from struct import unpack, calcsize
-
-import numpy, re
+import numpy as np
 
 COUPLINGS = ['AC', 'DC', 'GND']
-TRIG_CHANNELS = ['AUX','CH1','CH2','CH3','CH4','LINE']
+TRIG_CHANNELS = ['AUX', 'CH1', 'CH2', 'CH3', 'CH4', 'LINE']
 VERT_DIVISIONS = 8.0
 HORZ_DIVISIONS = 10.0
 SCALES = []
 
+
 class AgilentDSO91304AServer(GPIBManagedServer):
     name = 'Agilent Infiniium Oscilloscope'
-    deviceName = ['Agilent Technologies DSO91304A', 'KEYSIGHT TECHNOLOGIES DSO90804A']
-        
+    deviceName = ['Agilent Technologies DSO91304A',
+                  'KEYSIGHT TECHNOLOGIES DSO90804A']
+
     @setting(11, returns=[])
     def reset(self, c):
         dev = self.selectedDevice(c)
-        yield dev.write('*RST')
+        yield dev.query('*RST;*OPC?')
         # TODO wait for reset to complete
```

Comment:
With the addition of `*OPC?`, does this now wait until the rest completes? If so, you can remove the TODO.